code/game/objects/items/tools/mining_tools.dm

```diff
@@ -98,6 +86,238 @@
+				update_icon()
+			else
+				if(M)
+					fizzle_message(M)
+				return
+		else
+			if(M)
+				fizzle_message(M)
+				return
+	else
+		playsound(loc, 'sound/weapons/saberoff.ogg', 25)
+		powered = FALSE
+		if(M)
+			to_chat(M, "<span class='notice'>You switch [src] off. <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+		update_icon()
+
+/obj/item/tool/pickaxe/plasmacutter/proc/fizzle_message(mob/user)
+	var/fizzle = "<span class='warning'>The plasma cutter has inadequate charge remaining! Replace or recharge the battery.</span>"
+	playsound(src, 'sound/machines/buzz-two.ogg', 25, 1)
+	if(!cell)
+		to_chat(user, "<span class='warning'>[src]'s has no battery installed!</span>")
+	else if(!powered)
+		to_chat(user, "<span class='warning'>[src] is turned off!</span>")
+	else
+		to_chat(user, "<span class='warning'>[fizzle] <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+
+/obj/item/tool/pickaxe/plasmacutter/proc/start_cut(mob/user, name = "", atom/source)
+	eyecheck(user)
+	playsound(source, cutting_sound, 25, 1)
+	var/datum/effect_system/spark_spread/spark_system
+	spark_system = new /datum/effect_system/spark_spread()
+	spark_system.set_up(5, 0, source)
+	spark_system.attach(source)
+	spark_system.start(source)
+	to_chat(user, "<span class='notice'>You start cutting apart the [name] with [src]. <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+
+/obj/item/tool/pickaxe/plasmacutter/proc/cut_apart(mob/user, name = "", atom/source, charge_amount = charge_cost)
+	eyecheck(user)
+	playsound(source, cutting_sound, 25, 1)
+	var/datum/effect_system/spark_spread/spark_system
+	spark_system = new /datum/effect_system/spark_spread()
+	spark_system.set_up(5, 0, source)
+	spark_system.attach(source)
+	spark_system.start(source)
+	use_charge(user, charge_amount, FALSE)
+	to_chat(user, "<span class='notice'>You cut apart the [name] with [src]. <b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+
+/obj/item/tool/pickaxe/plasmacutter/proc/use_charge(mob/user, amount = charge_cost, mention_charge = TRUE)
+	cell.charge -= min(cell.charge, amount)
+	if(mention_charge)
+		to_chat(user, "<span class='notice'><b>Charge Remaining: [cell.charge]/[cell.maxcharge]</b></span>")
+	update_icon()
+	..()
+
+/obj/item/tool/pickaxe/plasmacutter/proc/calc_delay(mob/user)
+	var/final_delay = PLASMACUTTER_CUT_DELAY
+	if (istype(user))
+		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_ENGI)
+			user.visible_message("<span class='notice'>[user] fumbles around figuring out how to use [src].</span>",
+			"<span class='notice'>You fumble around figuring out how to use [src].</span>")
+			final_delay *= max(1, 4 + (user.mind.cm_skills.engineer * -1)) //Takes twice to four times as long depending on your skill.
+		else
+			final_delay -= min(PLASMACUTTER_CUT_DELAY,(user.mind.cm_skills.engineer - 3)*5)
```

Comment:
It's the `else` regarding `user.mind && user.mind.cm_skills` which makes this odd, since you are asking for those values here as well.

Reply:
If you want me to use a variable, just say so; I'll do that now.